src/LanguageServer.php

```diff
@@ -289,10 +324,48 @@ public function initialize(ClientCapabilities $capabilities, string $rootPath =
         });
     }
 
+    /**
+     * The initialized notification is sent from the client to the server after
+     * the client received the result of the initialize request but before the
+     * client is sending any other request or notification to the server.
+     *
+     * @return Promise
+     */
+    public function initialized(): Promise
+    {
+        return coroutine(function () {
+            if (!$this->rootPath) {
+                return;
+            }
+
+            // request configuration if it is supported
+            // support comes with protocol version 3.6.0
+            if ($this->clientCapabilities->workspace->configuration) {
+                $configurationitem = new ConfigurationItem();
+                $configurationitem->section = 'php';
+                $configuration = yield $this->client->workspace->configuration([$configurationitem]);
+                $options = $this->mapper->map($configuration[0], new Options());
```

Comment:
Shouldn't this be done by the `configuration` method?

Reply:
As long as we don't say, we will only use settings from the section "php", i would not do that, because I know that people will ask for including `files.exclude` setting (see https://github.com/felixfbecker/php-language-server/issues/159#issuecomment-306143750).